docs/api_reference/src/quantities.tex

```diff
@@ -131,136 +567,3373 @@
+
+template<typename T>
+concept @\defexposconceptnc{CastableNumber}@ =
+  @\exposconceptnc{CommonTypeWith}@<T, std::intmax_t> && @\exposconceptnc{ScalableNumber}@<std::common_type_t<T, std::intmax_t>>;
+
+template<typename T>
+concept @\defexposconceptnc{Scalable}@ =
+  @\exposconceptnc{CastableNumber}@<T> ||
+  (@\exposconceptnc{CastableNumber}@<@\exposidnc{actual-value-type-t}@<T>> &&
+   @\exposconceptnc{ScalableNumber}@<T, std::common_type_t<@\exposidnc{actual-value-type-t}@<T>, std::intmax_t>>);
+
+template<typename T>
+concept @\defexposconceptnc{WeaklyRegular}@ = std::@\stdconcept{copyable}@<T> && std::@\stdconcept{equality_comparable}@<T>;
+
+template<typename T>
+concept @\deflibconcept{Representation}@ = (is_scalar<T> || is_complex<T> || is_vector<T> || is_tensor<T>) &&
+                         @\exposconceptnc{WeaklyRegular}@<T> && @\exposconceptnc{Scalable}@<T>;
+
+template<typename T, quantity_character Ch>
+concept @\deflibconcept{RepresentationOf}@ =
+  @\libconcept{Representation}@<T> && ((Ch == quantity_character::scalar && is_scalar<T>) ||
+                        (Ch == quantity_character::complex && is_complex<T>) ||
+                        (Ch == quantity_character::vector && is_vector<T>) ||
+                        (Ch == quantity_character::tensor && is_tensor<T>));
+\end{itemdecl}
+
+\rSec1[qty.types]{Quantity types}
+
+\rSec2[qty.traits]{Traits}
+
+\rSec3[qty.val.traits]{Values}
+
+\begin{codeblock}
+namespace mp_units {
+
+template<typename Rep>
+struct @\libglobal{quantity_values}@ : std::chrono::duration_values<Rep> {
+  static constexpr Rep one() noexcept;
+};
+
+}
+\end{codeblock}
+
+\pnum
+The requirements on \tcode{std::chrono::duration_values<Rep>}\irefcpp{time.traits.duration.values}
+also apply to \tcode{quantity_values<Rep>}.
+
+\begin{itemdecl}
+static constexpr Rep one() noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{Rep(1)}.
+
+\pnum
+\remarks
+The value returned shall be the neutral element for multiplication\irefiev{102-01-19}.
+\end{itemdescr}
+
+\rSec3[qty.compat.traits]{Compatibility}
+
+\pnum
+The interfaces specified in this subclause
+are used by the quantity types\iref{term.quantity.type}
+to specify conversions with other types representing quantities.
+\ref{qty.chrono} implements them for \tcode{std::chrono::duration} and \tcode{std::chrono::time_point}.
+
+\begin{itemdecl}
+template<typename T, template<typename> typename Traits>
+concept @\defexposconceptnc{qty-like}@ = requires(const T& qty, const Traits<T>::rep& num) {
+  requires !@\exposidnc{is-specialization-of}@(^T, ^quantity);
+  requires !@\exposidnc{is-specialization-of}@(^T, ^quantity_point);
+  { Traits<T>::to_numerical_value(qty) } -> std::@\stdconcept{same_as}@<typename Traits<T>::rep>;
+  { Traits<T>::from_numerical_value(num) } -> std::@\stdconcept{same_as}@<T>;
+  { Traits<T>::explicit_import } -> std::same_as<const bool>;
+  { Traits<T>::explicit_export } -> std::same_as<const bool>;
```

Comment:
The actual expression type is `const bool &`

```suggestion
  { Traits<T>::explicit_import } -> std::convertible_to<bool>;
  { Traits<T>::explicit_export } -> std::convertible_to<bool>;
```

Reply:
I required it to be the same, as required of the boolean variable template traits.
Users don't need this freedom.

Reply:
Then you probably have to add `std::remove_cvref_t` or just `std::remove_reference_t`. Otherwise, it does not work.

Reply:
Makes sense.
How about this?
```suggestion
  { Traits<T>::explicit_import } -> std::same_as<const bool&>;
  { Traits<T>::explicit_export } -> std::same_as<const bool&>;
```

Reply:
Can we somehow prevent:

```cpp
  static constexpr const bool& explicit_import = false;
  static constexpr const bool& explicit_export = false;
```

Or should this be considered a valid solution?

Reply:
That seems more reasonable than `convertible_to<bool>`,
although it's possible to reject with:
```suggestion
  requires std::same_as<decltype(Traits<T>::explicit_import), const bool>;
  requires std::same_as<decltype(Traits<T>::explicit_export), const bool>;
```

Reply:
This seems to prevent references, indeed. However, I am not sure which option is the best here.

Reply:
The last option, which is the most restrictive.
It also stands out, in case LEWG wants to bikeshed.

Reply:
We could also go with _`boolean-testable`_.